lib/omniauth/strategies/auth0.rb

```diff
@@ -23,9 +24,9 @@ def initialize(app, *args, &block)
         @options.connection = args[4] unless args[4].nil?
 
         @options.client_options.site          = "https://#{options[:namespace]}"
-        @options.client_options.authorize_url = "https://#{options[:namespace]}/authorize"
-        @options.client_options.token_url     = "https://#{options[:namespace]}/oauth/token"
-        @options.client_options.userinfo_url  = "https://#{options[:namespace]}/userinfo"
+        @options.client_options.authorize_url = "https://#{options[:namespace]}/authorize?#{client_info_querystring}"
+        @options.client_options.token_url     = "https://#{options[:namespace]}/oauth/token?#{client_info_querystring}"
+        @options.client_options.userinfo_url  = "https://#{options[:namespace]}/userinfo?#{client_info_querystring}"
```

Comment:
* `/oauth/token`: please use request header instead of QS
* `/userinfo`: it's not neccesary